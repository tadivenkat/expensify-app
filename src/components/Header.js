import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogOut} from '../actions/auth';

export const Header = (props) => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <NavLink className="navbar-brand" to="/dashboard"><h1>Keilon Software Solutions Private Ltd</h1></NavLink>
            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
            </button>
            <div className="collapse navbar-collapse" id="navbarSupportedContent">
                <ul className="navbar-nav ml-auto">
                    <li className="nav-item"><NavLink to="/dashboard">Home</NavLink></li>
                    <li className="nav-item"><NavLink to="/create">Add Expense</NavLink></li>
                    <li className="nav-item"><button className="btn btn-primary btn-lg" onClick={props.startLogOut}><i className="fas fa-sign-out-alt"></i>Logout</button></li>
                </ul>
            </div>
        </nav>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        startLogOut : () => dispatch(startLogOut())
    }
};

export default connect(undefined, mapDispatchToProps)(Header);