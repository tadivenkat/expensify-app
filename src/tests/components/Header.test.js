import React from 'react';
import Header from '../../components/Header';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
//import ReactShallowRenderer from 'react-test-renderer/shallow';

test('Testing Header component', () => {
    // Testcase using enzyme
    const wrapper = shallow(<Header/>);
    //expect(wrapper.find('h1').text()).toBe('Keilon Software Solutions Private Ltd');
    expect(toJSON(wrapper)).toMatchSnapshot();
    
    // Testcase using react-test-renderer
    //const renderer = new ReactShallowRenderer();
    //renderer.render(<Header/>);
    //expect(renderer.getRenderOutput()).toMatchSnapshot();
});