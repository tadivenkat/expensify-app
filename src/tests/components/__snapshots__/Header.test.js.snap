// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/dashboard"
  >
    <h1>
      Keilon Software Solutions Private Ltd
    </h1>
  </a>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
      >
        <NavLink
          to="/dashboard"
        >
          Home
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <NavLink
          to="/create"
        >
          Add Expense
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <button
          className="btn btn-primary btn-lg"
          onClick={[MockFunction]}
        >
          <i
            class="fas fa-sign-out-alt"
          >
            Logout
          </i>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;
